File: BackEnd/src/main/java/io/ciudadlimpia/app/perfil/PerfilResponse.java
package io.ciudadlimpia.app.perfil;

import lombok.AllArgsConstructor;
import lombok.Getter;

@Getter
@AllArgsConstructor
public class PerfilResponse {
    private Long id;
    private String nombre;
    private String apellido;
    private String dni;
    private String email;
    private String fotoPerfil;
    private Integer puntosTotal;
    private NivelInfo nivel;
    private String siguienteNivel;
    private Integer totalReportes;
    private Integer reportesAtendidos;

    @Getter
    @AllArgsConstructor
    public static class NivelInfo {
        private Integer id;
        private String nombre;
        private String icono;
        private Integer puntosParaSiguienteNivel;
    }
}

File: AndroidStudio/app/src/main/java/com/nickrodriguez/ciudadlimpia/model/PerfilResponse.kt
package com.nickrodriguez.ciudadlimpia.model

data class PerfilResponse(
    val id: Long,
    val nombre: String,
    val apellido: String,
    val dni: String,
    val email: String,
    val fotoPerfil: String?,
    val puntosTotal: Int,
    val nivel: NivelInfo,
    val siguienteNivel: String,
    val totalReportes: Int,
    val reportesAtendidos: Int
)
data class NivelInfo(
    val id: Int,
    val nombre: String,
    val icono: String,
    val puntosParaSiguienteNivel: Int
)